sklearn/covariance/_shrunk_covariance.py

```diff
@@ -484,12 +451,16 @@ def fit(self, X, y=None):
         # Not calling the parent object to fit, to avoid computing the
         # covariance matrix (and potentially the precision)
         X = self._validate_data(X)
+        if X.shape[0] == 1:
+            warnings.warn(
+                "Only one sample available. You may want to reshape your data array"
+            )
```

Comment:
It seems that we were supporting 1D array in the function. Therefore, we can let go the 1D vector and reshape and raise the warning afterwards.
```suggestion
        X = self._validate_data(X, ensure_2d=False)
        if X.dim == 1:
            X = X.reshape(1, -1)
            warnings.warn(
                "Only one sample available. You may want to reshape your data array"
            )
```